include/revng/Model/Register.h

```diff
@@ -8,688 +8,12 @@
 
 #include "revng/ADT/KeyedObjectTraits.h"
 #include "revng/Model/Architecture.h"
+#include "revng/Model/GeneratedRegister.h"
+#include "revng/Model/GeneratedRegisterState.h"
```

Comment:
As previously, why do we need this?
Also, missing `_late` inclusion.

Reply:
Well, we need `GeneratedRegister.h` (now `Generated/Early/Register.h`) because `inline llvm::StringRef getRegisterName(Values V)` refers to `model::Register::Values`.
Same goes for `isYesOrDead`, it refers to `model::RegisterState::Values`. I can create a separate `RegisterState.h` if you prefer.
As I was saying in the other reply, enums do not need (nor have) a `_late` file to include.

Reply:
Can't you include `revng/Model/RegisterState.h`?
Maybe `revng/Model/GeneratedRegister.h` should do that.

Reply:
I created a `RegisterState.h`, now `Register.h` has nothing related to `RegisterState` (both generated and manually written)